pkg/scheduler/api/validation/validation.go

```diff
@@ -66,6 +74,44 @@ func ValidatePolicy(policy schedulerapi.Policy) error {
 	return utilerrors.NewAggregate(validationErrors)
 }
 
+// validatePriorityRedeclared checks if any custom priorities have been declared multiple times in the policy config
+// by examining the specified priority arguments
+func validatePriorityRedeclared(priorities map[string]schedulerapi.PriorityPolicy, priority schedulerapi.PriorityPolicy) error {
+	var priorityType string
+	if priority.Argument.LabelPreference != nil {
+		priorityType = "LabelPreference"
+	} else if priority.Argument.RequestedToCapacityRatioArguments != nil {
+		priorityType = "RequestedToCapacityRatioArguments"
+	} else if priority.Argument.ServiceAntiAffinity != nil {
+		priorityType = "ServiceAntiAffinity"
+	} else {
+		return fmt.Errorf("No priority arguments set for priority %s", priority.Name)
+	}
+	if existing, alreadyDeclared := priorities[priorityType]; alreadyDeclared {
+		return fmt.Errorf("Priority %s is redeclared (was %+v)", priority.Name, existing)
+	}
+	priorities[priorityType] = priority
+	return nil
+}
+
+// validatePredicateRedeclared checks if any custom predicates have been declared multiple times in the policy config
+// by examining the specified predicate arguments
+func validatePredicateRedeclared(predicates map[string]schedulerapi.PredicatePolicy, predicate schedulerapi.PredicatePolicy) error {
+	var predicateType string
+	if predicate.Argument.LabelsPresence != nil {
```

Comment:
I think you need to check whether predicate.Argument is nil or not.

Currently `nil` argument is valid if the name of the predicate matches a predefined one (I don't know if this is desired or not), see
https://github.com/kubernetes/kubernetes/blob/e1f86e346062e36aa03fc53ea4fb05d4a61ef458/pkg/scheduler/algorithm_factory.go#L281